flink-table-store-hive/src/main/java/org/apache/flink/table/store/mapred/TableStoreInputFormat.java

```diff
@@ -0,0 +1,180 @@
+ * bucket) and creates {@link RecordReader} for each split.
+ */
+public class TableStoreInputFormat implements InputFormat<Void, RowDataContainer> {
+
+    @Override
+    public InputSplit[] getSplits(JobConf jobConf, int numSplits) throws IOException {
+        FileStoreImpl store = getFileStore(jobConf);
+        FileStoreScan scan = store.newScan();
+        List<TableStoreInputSplit> result = new ArrayList<>();
+        for (Map.Entry<BinaryRowData, Map<Integer, List<DataFileMeta>>> pe :
+                scan.plan().groupByPartFiles().entrySet()) {
+            for (Map.Entry<Integer, List<DataFileMeta>> be : pe.getValue().entrySet()) {
+                BinaryRowData partition = pe.getKey();
+                int bucket = be.getKey();
+                String bucketPath =
+                        store.pathFactory()
+                                .createDataFilePathFactory(partition, bucket)
+                                .toPath("")
+                                .toString();
+                TableStoreInputSplit split =
+                        new TableStoreInputSplit(
+                                store.partitionType(),
+                                store.keyType(),
+                                store.valueType(),
+                                partition,
+                                bucket,
+                                be.getValue(),
+                                bucketPath);
+                result.add(split);
+            }
+        }
+        return result.toArray(new TableStoreInputSplit[0]);
+    }
+
+    @Override
+    public RecordReader<Void, RowDataContainer> getRecordReader(
+            InputSplit inputSplit, JobConf jobConf, Reporter reporter) throws IOException {
+        FileStore store = getFileStore(jobConf);
+        FileStoreRead read = store.newRead();
+        TableStoreInputSplit split = (TableStoreInputSplit) inputSplit;
+        org.apache.flink.table.store.file.utils.RecordReader wrapped =
+                read.withDropDelete(true)
+                        .createReader(split.partition(), split.bucket(), split.files());
+        long splitLength = split.getLength();
+        return new JobConfWrapper(jobConf).getPrimaryKeyNames().isPresent()
+                ? new TableStorePkRecordReader(wrapped, splitLength)
+                : new TableStoreCountRecordReader(wrapped, splitLength);
+    }
+
+    private FileStoreImpl getFileStore(JobConf jobConf) {
+        JobConfWrapper wrapper = new JobConfWrapper(jobConf);
+
+        String catalogName = wrapper.getCatalogName();
+        String dbName = wrapper.getDbName();
+        String tableName = wrapper.getTableName();
+        ObjectIdentifier identifier = ObjectIdentifier.of(catalogName, dbName, tableName);
+
+        Configuration fileStoreOptions = new Configuration();
+        fileStoreOptions.setString(FileStoreOptions.PATH, wrapper.getLocation());
```

Comment:
The location should be table location.
We can get rid of `catalogName` and `dbName` and `tableName`